@here/harp-mapview/lib/MapView.ts

```diff
@@ -2418,11 +2418,12 @@ export class MapView extends THREE.EventDispatcher {
         elevationRangeSource: ElevationRangeSource,
         elevationProvider: ElevationProvider
     ) {
-        // Try to remove incase this method was already called, will do nothing if it doesn't exist.
-        this.removeDataSource(elevationSource);
-        this.addDataSource(elevationSource);
+        // Add as datasource if it was not added before
+        const isPresent = this.m_tileDataSources.some(ds => ds === elevationSource);
```

Comment:
you are slightly modifying the behaviour of the API, because now the user can no longer swap between two elevation sources, once they call this method, the MapView can only use that specific ElevationSource. I don't know if there is a usecase to swap elevation sources, but because it is currently possible in the API, I think it should at least be documented that once an elevatoin source is set, it can't be changed.

I think this is

Reply:
Ok, but that wouldn't have worked before either. The code would have removed the `elevationSource` that was given as a parameter and than add it again. For switching the source we may want to remove the previous `mapView.m_elevationRangeSource`.

Reply:
Oh you are right, I misread, resolving.